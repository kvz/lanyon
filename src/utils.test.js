const utils = require('./utils')
const oneLine = require('common-tags/lib/oneLine')
// const debug      = require('depurar')('sut')
// const sut           = utils
// const cacheDir      = 'CACHEDIR'
// const projectDir    = 'PROJECTDIR'
// const lanyonVersion = 'LANYONVERSION'

describe('utils', () => {
  test('dockerString', () => {
    const p = {
      runtime: {
        contentBuildDir: '/Users/kvz/code/project/_site',
        cacheDir       : '/Users/kvz/code/project/.lanyon',
        projectDir     : '/Users/kvz/code/project',
        dockerImage    : 'kevinvz/lanyon:v1.0.109',
        lanyonEnv      : 'production',
      },
    }
    const res = utils.dockerString('ls', p)
    expect(res).toMatch(new RegExp(oneLine`
      ^docker run
      --rm
      --interactive
      --env "JEKYLL_UID=\\d+"
      --env "JEKYLL_ENV=production"
      --workdir /Users/kvz/code/project/.lanyon
      --volume /Users/kvz/code/project:/Users/kvz/code/project kevinvz/lanyon:v1.0.109 ls$
    `))
  })
})
