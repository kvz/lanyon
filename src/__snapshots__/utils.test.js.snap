// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils dockerCmd should add custom flags 1`] = `"docker run FLAGS --user $(id -u) --workdir CACHEDIR --volume CACHEDIR:CACHEDIR --volume PROJECTDIR:PROJECTDIR kevinvz/lanyon:LANYONVERSION CMD"`;
